mitodl/mitxonline
# Adding Google Sheets Refunds functionality to mitxonline

#### Pre-Flight checklist


#### What are the relevant tickets?
Fix #669

#### What's this PR do?
Installs mitol-google-sheets, and mitol-google-sheets-refunds libraries
Adds implementation of the refund process 

When google sheet is configured for processing refunds, the following process takes place:
1. Staff enters refund information in the configured google sheet
2. An scheduled task in MITx Online picks up the new change and  updates the order status to reflect the refund with a refund transaction
3. MITx Online calls the CyberSource API to issue the refund to the learner
4. MITx Online updates the google sheet indicating that the refund has been processed successfully or with an indication of failure

#### How should this be manually tested?

Follow the instructions [here](https://github.com/mitodl/ol-django/blob/85bea3ec5da01180ef943deb89b14d1463eb7c21/src/mitol/google_sheets/README.md) for developer setup.
When you Google Drive is set up correctly, proceed to testing:

1. Fill out and submit the spreadsheet request form. You need to link the refund form responses form to output to the spreadsheet. You can do that by opening the form 
and Responses->Settings->Select response destination-> Select existing spreadsheet.
When you fill out the form it will create a new worksheet, called something like "Form Response 1". You can
rename this tab to "Refund Form Response".
In the "Refund Form Response" sheet make sure that the "Timestamp" column format is set to "Date" and not to "Date Time".
The main google worksheet gets updated by the "Refund Response" worksheet by the following query:

```markdown
={QUERY({'Refund Form Response'!A2:G, ARRAYFORMULA(if(isblank('Refund Form Response'!A2:A),"",ROW('Refund Form Response'!A2:G)))},"SELECT Col8, Col1, Col3, Col4, Col2, Col5, Col6, Col7",0)}
```
Add this query to the first data row and column (4:A).

2. Run the management command to process the sheet:
`./manage.py process_refund_requests -i "<spreadsheet id>"`. This should update the "Order Completed Date" column for the row you added.
